docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
 
-<tbody>
-<tr>
-<td>
+The `Context` type from `graphql-ws` is where information about your subscription server and its requests resides. You can access incoming parameters from a client `subscription` request by checking the `Context.connectionParams`.
 
-###### `connectionParams`
+Below is an example of the common use case of extracting an authentication token from a client `subscription` request and using it to find the current user:
 
-`Object`
-
-</td>
-<td>
-
-**Passed to `onConnect` only.**
-
-An object containing parameters included in the request, such as an [authentication token](#example-authentication-with-onconnect).
-
-For details, see [Authenticate over WebSocket](https://www.apollographql.com/docs/react/data/subscriptions/#4-authenticate-over-websocket-optional) in the Apollo Client documentation.
-
-</td>
-</tr>
-
-<tr>
-<td>
-
-###### `webSocket`
-
-`WebSocket`
-</td>
-<td>
-
-The connecting or disconnecting [`WebSocket`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket).
+```ts
+const getDynamicContext = async (ctx, msg, args) => {
+  // ctx is the graphql-ws Context where connectionParams live
+ if (ctx.connectionParams.authentication) {
+    const currentUser = await findUser(connectionParams.authentication);
+    return { currentUser };
+  }
+  // Otherwise let our resolvers know we don't have a current user
+  return { currentUser: null };
+};
 
-</td>
-</tr>
+useServer(
+  {
+    schema,
+    // Adding a context property lets you add data to your GraphQL operation context
+     context: (ctx, msg, args) => {
+       // Returning an object will add that information to our
+       // GraphQL context, which all of our resolvers have access to.
+      return getDynamicContext(ctx, msg, args);
+    },
+  },
+  wsServer,
+);
+```
+Putting it all together, if the `useServer.context` function returns an _object_ that object is passed to the GraphQL operation `context`, which is available to your resolvers.
 
-<tr>
-<td>
+Note that the `context` option is called once per `subscription` request, not once per event emission. Meaning that in the above example, every time a client sends a `subscription` operation we would check their authentication token.
 
-###### `context`
+### `onConnect` and `onDisconnect`
 
-`ConnectionContext`
-</td>
-<td>
+Expanding on the above, you can add additional options for the subscription server to execute whenever a client connects (`onConnect`) or disconnects (`onDisconnect`).
 
-Context object for the WebSocket connection. This is _not_ the `context` object for the associated subscription operation.
+Defining an `onConnect` function lets you reject a particular incoming connection by returning false or by throwing an exception. This can be helpful if you'd like to check authentication when a client first connects to your subscription server.
```

Comment:
"lets you" -> "enables you to"

Codefont `false`

"you'd like to" -> "you want to"